crates/core/src/subscription/subscription.rs

```diff
@@ -93,48 +118,38 @@ impl QuerySet {
         let mut output = DatabaseUpdate { tables: vec![] };
         let mut seen = HashSet::new();
 
-        for query in &self.0 {
-            for table in database_update.tables.iter().cloned() {
-                for q in query.queries_of_table_id(&table) {
-                    if let Some(result) = run_query(relational_db, tx, &q, auth)?
-                        .into_iter()
-                        .find(|x| !x.data.is_empty())
-                    {
-                        let pos_op_type = result.head.find_pos_by_name(OP_TYPE_FIELD_NAME).unwrap_or_else(|| {
-                            panic!(
-                                "failed to locate `{OP_TYPE_FIELD_NAME}` on `{}`. fields: {:?}",
-                                result.head.table_name,
-                                result.head.fields.iter().map(|x| &x.field).collect::<Vec<_>>()
-                            )
+        for expr in self {
+            use query::Supported::*;
+            match query::classify(expr) {
+                None => log::warn!("invalid query expression for incremental evaluation"),
```

Comment:
It is probably preferable to classify the queries when subscribing, and return an error if there are unsupported ones. Threading the tag through would make the diff much larger though, so I punted on that for now.